optimism/src/main.rs

```diff
@@ -40,12 +40,113 @@ pub fn main() -> ExitCode {
 
     env_logger::Builder::from_env(env_logger::Env::default().default_filter_or("info")).init();
 
-    let mut env = witness::Env::<ark_bn254::Fq>::create(cannon::PAGE_SIZE as usize, state, po);
+    let domain_size = 1 << 15;
+
+    let domain =
+        kimchi::circuits::domains::EvaluationDomains::<ark_bn254::Fr>::create(domain_size).unwrap();
+
+    let srs = {
+        use ark_ff::UniformRand;
+
+        // Trusted setup toxic waste
+        let x = ark_bn254::Fr::rand(&mut rand::rngs::OsRng);
+
+        let mut srs = poly_commitment::pairing_proof::PairingSRS::create(x, domain_size);
+        srs.full_srs.add_lagrange_basis(domain.d1);
+        srs
+    };
+
+    let mut env = witness::Env::<ark_bn254::Fr>::create(cannon::PAGE_SIZE as usize, state, po);
+
+    let mut accumulator = proof::ProofInputs::<
+        ark_ec::short_weierstrass_jacobian::GroupAffine<ark_bn254::g1::Parameters>,
+    >::default();
+
+    let new_chunk = || proof::WitnessColumns {
```

Comment:
what is exactly a chunk in this context? chunks as in kimchi in case the circuit was larger than the domain size?

Reply:
It's a pre-folding witness. The domain size we're using is 2^15, so a 'chunk' is 2^15 constraints/rows.